network/onchain-discovery/src/lib.rs

```diff
@@ -64,41 +62,42 @@ pub mod service;
 pub mod types;
 
 /// Query our own storage for the latest discovery set and epoch change proof.
-async fn storage_query_discovery_set(
+fn storage_query_discovery_set(
     libra_db: Arc<dyn DbReader>,
     req_msg: QueryDiscoverySetRequest,
-) -> Result<(QueryDiscoverySetRequest, QueryDiscoverySetResponseWithEvent)> {
-    let storage_req_msg = Into::<UpdateToLatestLedgerRequest>::into(req_msg.clone());
-
-    let res_msg = libra_db
-        .update_to_latest_ledger(
-            storage_req_msg.client_known_version,
-            storage_req_msg.requested_items,
-        )
-        // .await
+) -> Result<(QueryDiscoverySetRequest, Box<QueryDiscoverySetResponse>)> {
+    // TODO(philiphayes): how to deal with partial epoch change proof?
```

Comment:
I'd ping @lightmark

Reply:
we'll handle this in a follow up, since it requires some refactoring of trusted_state and libradb